libethereum/Client.cpp

```diff
@@ -651,6 +651,8 @@ void Client::onPostStateChanged()
 void Client::startMining()
 {
 	m_wouldMine = true;
+	DEV_WRITE_GUARDED(x_working)
```

Comment:
would need `m_preMine` to be read locked at the least.

really this logic should be in `rejigMining`, though.